plugins/inputs/prometheus/prometheus.go

```diff
@@ -109,11 +125,25 @@ func (p *Prometheus) GetAllURLs() ([]URLAndAddress, error) {
 
 		allURLs = append(allURLs, URLAndAddress{URL: URL, OriginalURL: URL})
 	}
+	p.lock.Lock()
+	defer p.lock.Unlock()
+	// loop through all pods scraped via the prometheus annotation on the pods
+	for _, pod := range p.KubernetesPods {
+		URL, err := url.Parse(pod.url)
+		if err != nil {
+			log.Printf("prometheus: Could not parse url %s, skipping it. Error: %s", pod.url, err)
+			continue
+		}
+		podURL := p.AddressToURL(URL, URL.Hostname())
+		allURLs = append(allURLs, URLAndAddress{URL: podURL, Address: URL.Hostname(), OriginalURL: URL, Tags: pod.tags})
```

Comment:
If we merge `URLAndAddress` with `Target` then we can do this logic in kubernetes.go and all we need to do here is add the items to the lists.